### PR TITLE
[Enchance] Update FAQ docs

### PR DESCRIPTION
**Pull request by hhaAndroid**
## Motivation

Update FAQ docs

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -83,8 +83,39 @@ We list some common troubles faced by many users and their corresponding solutio
...
+            self.conv1_stride = 1
+            self.conv2_stride = stride
+        else:
+            self.conv1_stride = stride
+            self.conv2_stride = 1
+    ```
+
+- ResNeXt parameter description
+
+    ResNeXt comes from the paper `Aggregated Residual Transformations for Deep Neural Networks`. It introduces grouped convolution and uses “cardinality” to control the number of groups to achieve a balance between accuracy and complexity. It controls the basic width and grouping parameters of the internal Bottleneck module through two hyperparameters `baseWidth` and `cardinality`. The configuration name in MMDetection is `mask_rcnn_x101_64x4d_fpn_mstrain-poly_3x_coco.py`, where `mask_rcnn` represents the algorithm using Mask R-CNN, `x101` represents the backbone network using ResNeXt-101, and `64x4d` represents the Botleneck basic width is 4, which is divided into 64 group.
```
Comment (ZwwWayne) [inline docs/faq.md:117]: add paper link at the place of paper name

Comment (ZwwWayne) [inline docs/faq.md:117]: 64x4d represents that the bottleneck block has 64 group and each group has basic width of 4

File: docs/faq.md [hunk 2]
```diff
@@ -83,8 +83,39 @@ We list some common troubles faced by many users and their corresponding solutio
...
+
+- Backbone network BatchNorm layer eval mode description
```
Comment (ZwwWayne) [inline docs/faq.md:119]: Make the title more direc, e.g., About norm_eval option in backbone

File: docs/faq.md [hunk 3]
```diff
@@ -83,8 +83,39 @@ We list some common troubles faced by many users and their corresponding solutio
     1. This error indicates that your module has parameters that were not used in producing loss. This phenomenon may be caused by running different branches in your code in DDP mode.
     2. You can set ` find_unused_parameters = True` in the config to solve the above problems or find those unused parameters manually.
 
+- Save the best model
+
+    It can be turned on by configuring `evaluation = dict(save_best=‘auto’)`. In the case of the `auto` parameter, the first key in the returned evaluation result will be used as the basis for selecting the best model. You can also directly set the key in the evaluation result to manually set it, for example, `evaluation = dict(save_best='mAP' )`.
+
+- Usage of EMA Hook in Resume
+
+    If you use `EMA Hook` in training, you can't use command line parameters to restore model parameters during resume, you can only do so by modifying the `resume_from` field in the configuration file. This is because the implementation of `EMA Hook` is quite special, and the weights need to be loaded again during initialization.
```
Comment (ZwwWayne) [inline docs/faq.md:92]: It is a little bit confusing. what command line parameter cannot be used to restore the model? do you mean `--resume-from`?  What is the difference from `--resume-from` and `--cfg-options resume_from`? quite special is also vague

File: docs/faq.md [hunk 4]
```diff
@@ -83,8 +83,39 @@ We list some common troubles faced by many users and their corresponding solutio
...
+
 ## Evaluation
 
 - COCO Dataset, AP or AR = -1
     1. According to the definition of COCO dataset, the small and medium areas in an image are less than 1024 (32\*32), 9216 (96\*96), respectively.
     2. If the corresponding area has no object, the result of AP and AR will set to -1.
+
+## Model
+
+- ResNet style parameter description
```
Comment (ZwwWayne) [inline docs/faq.md:102]: `style` in ResNet

File: docs/faq.md [hunk 5]
```diff
@@ -83,8 +83,39 @@ We list some common troubles faced by many users and their corresponding solutio
...
+
+    ResNet `style` optional parameters include `pytorch` and `caffe`, the difference lies in the Bottleneck module. Bottleneck is a stacking structure of `1x1-3x3-1x1`. In the case of sampling with stride=2, the stride parameter of `caffe` mode is placed on the first `1x1` convolution, while in `pyorch` mode, stride is placed on the second `3x3` convolution. The sample code is as follows:
```
Comment (ZwwWayne) [inline docs/faq.md:104]: The `style` parameter in ResNet allows either `pytorch` or `caffe` style. It indicates the difference in the Bottleneck module.

File: docs/faq.md [hunk 6]
```diff
@@ -83,8 +83,39 @@ We list some common troubles faced by many users and their corresponding solutio
...
+            self.conv1_stride = 1
+            self.conv2_stride = stride
+        else:
+            self.conv1_stride = stride
+            self.conv2_stride = 1
+    ```
+
+- ResNeXt parameter description
+
+    ResNeXt comes from the paper [`Aggregated Residual Transformations for Deep Neural Networks`](https://arxiv.org/abs/1611.05431). It introduces grouped convolution and uses “cardinality” to control the number of groups to achieve a balance between accuracy and complexity. It controls the basic width and grouping parameters of the internal Bottleneck module through two hyperparameters `baseWidth` and `cardinality`. The configuration name in MMDetection is `mask_rcnn_x101_64x4d_fpn_mstrain-poly_3x_coco.py`, where `mask_rcnn` represents the algorithm using Mask R-CNN, `x101` represents the backbone network using ResNeXt-101, and `64x4d` represents that the bottleneck block has 64 group and each group has basic width of 4.
```
Comment (ZwwWayne) [inline docs/faq.md:117]: grouped convolution -> group

File: docs/faq.md [hunk 7]
```diff
@@ -83,8 +83,39 @@ We list some common troubles faced by many users and their corresponding solutio
...
+
+-  norm_eval option in backbone
```
Comment (ZwwWayne) [inline docs/faq.md:119]: `norm_eval` in backbone

File: docs/faq.md [hunk 8]
```diff
@@ -83,8 +83,39 @@ We list some common troubles faced by many users and their corresponding solutio
     1. This error indicates that your module has parameters that were not used in producing loss. This phenomenon may be caused by running different branches in your code in DDP mode.
     2. You can set ` find_unused_parameters = True` in the config to solve the above problems or find those unused parameters manually.
 
+- Save the best model
+
+    It can be turned on by configuring `evaluation = dict(save_best=‘auto’)`. In the case of the `auto` parameter, the first key in the returned evaluation result will be used as the basis for selecting the best model. You can also directly set the key in the evaluation result to manually set it, for example, `evaluation = dict(save_best='mAP' )`.
+
+- Usage of EMA Hook in Resume
+
+    If you use `EMA Hook` in training, you can't use command line parameters such as `--resume-from` and `--cfg-options resume_from`to restore model parameters during resume, you can only do so by modifying the `resume_from` field in the configuration file. This is because EMA Hook cannot obtain command line parameters, and thus cannot implement the weight reload function.
```
Comment (ZwwWayne) [inline docs/faq.md:92]: May give an example, still feel confused about the `resume_from` field in the configuration file`

File: docs/faq.md [hunk 9]
```diff
@@ -83,8 +84,51 @@ We list some common troubles faced by many users and their corresponding solutio
...
 
+- Save the best model
+
+    It can be turned on by configuring `evaluation = dict(save_best=‘auto’)`. In the case of the `auto` parameter, the first key in the returned evaluation result will be used as the basis for selecting the best model. You can also directly set the key in the evaluation result to manually set it, for example, `evaluation = dict(save_best='mAP' )`.
+
+- Usage of EMA Hook in Resume
+
+    If you use `EMA Hook` in training, you can't just use command line parameters such as `--resume-from` and `--cfg-options resume_from`to restore model parameters during resume e.g. `python tools/train.py configs/yolox/yolox_s_8x8_300e_coco.py --resume-from ./work_dir/yolox_s_8x8_300e_coco/epoch_x.pth `. Since EMA Hook needs to reload the weights, taking the `yolox_s` algorithm as an example, you can do this by doing the following:
+
+    ```shell
```
Comment (ZwwWayne) [inline docs/faq.md:95]: python rather than shell

File: docs/faq.md [hunk 10]
```diff
@@ -83,8 +84,51 @@ We list some common troubles faced by many users and their corresponding solutio
     1. This error indicates that your module has parameters that were not used in producing loss. This phenomenon may be caused by running different branches in your code in DDP mode.
     2. You can set ` find_unused_parameters = True` in the config to solve the above problems or find those unused parameters manually.
 
+- Save the best model
+
+    It can be turned on by configuring `evaluation = dict(save_best=‘auto’)`. In the case of the `auto` parameter, the first key in the returned evaluation result will be used as the basis for selecting the best model. You can also directly set the key in the evaluation result to manually set it, for example, `evaluation = dict(save_best='mAP' )`.
+
+- Usage of EMA Hook in Resume
```
Comment (ZwwWayne) [inline docs/faq.md:91]: Resume training with EMA Hook

File: docs/faq.md [hunk 11]
```diff
@@ -83,8 +84,51 @@ We list some common troubles faced by many users and their corresponding solutio
...
+
+    If you use `EMA Hook` in training, you can't just use command line parameters such as `--resume-from` and `--cfg-options resume_from`to restore model parameters during resume e.g. `python tools/train.py configs/yolox/yolox_s_8x8_300e_coco.py --resume-from ./work_dir/yolox_s_8x8_300e_coco/epoch_x.pth `. Since EMA Hook needs to reload the weights, taking the `yolox_s` algorithm as an example, you can do this by doing the following:
```
Comment (ZwwWayne) [inline docs/faq.md:93]: EMA Hook -> ExpMomentumEMAHook or xxx, use the exact module name

Comment (ZwwWayne) [reply docs/faq.md:93]: and `--cfg-options resume_from`to -> nor `--cfg-options resume_from` to

File: docs/faq.md [hunk 12]
```diff
@@ -83,8 +84,51 @@ We list some common troubles faced by many users and their corresponding solutio
...
 
 - COCO Dataset, AP or AR = -1
     1. According to the definition of COCO dataset, the small and medium areas in an image are less than 1024 (32\*32), 9216 (96\*96), respectively.
     2. If the corresponding area has no object, the result of AP and AR will set to -1.
+
+## Model
+
+- `style` in ResNet
+
+    The `style` parameter in ResNet allows either `pytorch` or `caffe` style. It indicates the difference in the Bottleneck module. Bottleneck is a stacking structure of `1x1-3x3-1x1`. In the case of sampling with stride=2, the stride parameter of `caffe` mode is placed on the first `1x1` convolution, while in `pyorch` mode, stride is placed on the second `3x3` convolution. The sample code is as follows:
```
Comment (ZwwWayne) [inline docs/faq.md:117]: The `style` parameter in ResNet allows either `pytorch` or `caffe` style. It indicates the difference in the Bottleneck module. Bottleneck is a stacking structure of `1x1-3x3-1x1` convolutional layers. In the case of `caffe` mode,  the  convolution layer with `stride=2` is the first `1x1` convolution, while in `pyorch` mode,it is the second `3x3` convolution has `stride=2`.  A sample code is as below:

File: docs/faq.md [hunk 13]
```diff
@@ -83,8 +84,51 @@ We list some common troubles faced by many users and their corresponding solutio
...
+            self.conv1_stride = 1
+            self.conv2_stride = stride
+        else:
+            self.conv1_stride = stride
+            self.conv2_stride = 1
+    ```
+
+- ResNeXt parameter description
+
+    ResNeXt comes from the paper [`Aggregated Residual Transformations for Deep Neural Networks`](https://arxiv.org/abs/1611.05431). It introduces  group and uses “cardinality” to control the number of groups to achieve a balance between accuracy and complexity. It controls the basic width and grouping parameters of the internal Bottleneck module through two hyperparameters `baseWidth` and `cardinality`. The configuration name in MMDetection is `mask_rcnn_x101_64x4d_fpn_mstrain-poly_3x_coco.py`, where `mask_rcnn` represents the algorithm using Mask R-CNN, `x101` represents the backbone network using ResNeXt-101, and `64x4d` represents that the bottleneck block has 64 group and each group has basic width of 4.
```
Comment (ZwwWayne) [inline docs/faq.md:130]: An example configuration name in MMDetection is `mask_rcnn_x101_64x4d_fpn_mstrain-poly_3x_coco.py`, where `mask_rcnn` represents the algorithm using Mask R-CNN, `x101` represents the backbone network using ResNeXt-101, and `64x4d` represents that the bottleneck block has 64 group and each group has basic width of 4.

**Review by ZwwWayne — APPROVED** [review]
